# Consensus Identity Index
Calculating Consensus identity index to set of sequences.

The program requires a consensus and a file with sequences. All sequences should be aligned and provided in the FASTA-format.

General description you can find in this article:

> Waye JS, Willard HF. Chromosome-specific alpha satellite DNA: nucleotide sequence analysis of the 2.0 kilobasepair repeat from the human X chromosome. Nucleic Acids Res. 1985 Apr 25;13(8):2731-43.
PubMed PMID: 2987865; PubMed Central PMCID: PMC341190.


### Required parameters:

`-v cons="consensus-align-seq.fas"` — FASTA file with consensus sequense(s), for example:
```fasta
>X02418.1_consX
AATCTGCAAGTGGATATTTGGACCNCTTTGAGGNNTTCGTTGGAAACGGNAATATCTTCACATAAAAACTAAACAGAAGCATTCTCAGAAACTTCTTTGTGATGATTGCATTCAACTCACAGAGTTGAACATTCCTTTTGATAGAGCAGTTTNGAAACACTCTTTTTGTAG
```

and `subject-align-seq.fas` — FASTA file with subject sequense(s), something like this:
```fasta
>X02418.1_X1
GATCCGCAAG-GGATATTTGGACCTCTTTGAAGGTTTCGTTGGAAACGGGATAATCTTCACCTAAAAGCTAAACGGAAGCATTCTCAGAAACTTCTTTGGGATGTTTGCATTCACCTCACAGAGTTGAAGTTTCCCTTTGATAGCGCAGTTT-GACACAC--TTTTTCTAC
>X02418.1_X2
AATGTGCAAGTGGCTATTTAGCGGGCTTGGAGGACTGTGTTGGAAAAGGAAATATCTTCTCCTAAAAACGACATAGAAGCATTCTCAGAAACTGCTCTGTGATGATTGCATTCAACTCCCAGAGTTGAACATTCCTTTTGATAGAGCAGTTTGCAAACACTCTTTTTGTAG
>X02418.1_X3
AATCTGCAAGTGGAGATTTGGACCGCTTTGAGGCCTGTGGTAGTGAAGGAAAGAACTTCATATAAAAACCAGACGGTAGCACTCTCAGAAAATTCTTTGTGACGATGGAGTTTAACTCAGGGAGCTGAACATTCGTTATGATGGAGCAGATTCCAAACACACGTTTTGTAG
>X02418.1_X4
AATCTGCGAGGGGATATTTGGAC-TCTCTCTGAGGTTTGTTGGAAACGGGATCACCTTCCCATAACTG---AACGGAAGCAAACTCAGAACATTCTTTGTGATGTTTGTATTCAACTCACAGAGTTGAACCTTCCTTT-GATAGTTCAGGTTTGCAACACCCTTGTAGTAG
>X02418.1_X5
AATCTGCAAGTGTATATTTTGACCACTTTGTAGCCTTCGTTTGAAACGTCTATATCTTCACATCAAACCTAGACAGAAGCATTCTCAGAAAGTTTTCTGCGATGACTGCATTCCACTCACAGAGTTGAACAATCCTTTTGATGGAGCAGTTTTGAAACCCTCTTTCTTTGG
>X02418.1_X6
AATCTGCAAGGGGATATGTGGACCTCTTTGAAGATTTCACTGGAAACGGGATCATCTTCACATAAAAACTAAACAGAAGCATTCTCGGAAACTATTTTGTGATGTTTGTATTCAACTCCCAGAGTTGAACTTTCCTTTTGAAAGAGCAGCTATGAAACACTCTTTTTCGAG
>X02418.1_X7
AATCTGCAAGTGGACGTTTGGAGGGCTTTGAGGC-TGTGGTGGAAAAGGAAATATCTTCACACAAAAACCAGATAGAAGCATTCTCAGAAACGACTTTGTGAGGATGGCATTCAACTACTGGAGTTGA-CAATCCTATTGATAGAGCAGATTGGAATCACTCTTTTTGTAG
>X02418.1_X8
AATCTGCAAATGGAGATTTGGACTGCTTTGAGGCCTACGGTGGTACAGGAAGGAACTTCATATAAAAGGCAAACGGAAGCATTCTCAGAATATTCTTTGTGATGATGGAGTTTCACTCACAGAGCTGAACATGCCTTTTGATGGAGCAGTTTCCAAATACACTTTTGGTAG
>X02418.1_X9
AATCTGCAGGTGGATATTTGGAGCTCTCTGAGGATTTCGTTGGAAACGGGAATAATTTCCCATAACTA---AACACAAACAATCTGAGAAAGTTCTTCATGATGAATGCATTTAACTCGCAGAGATGAACCTGCCTTT-GAGAGTTCAGGTTCGAAACACTCTTTCTGTAG
>X02418.1_X10
AATCTGCAAGTGGATATTTGGACCACTGGGT-GCCTTCGTTCGAAACGGGTATATGCTCACGTAAAAACTAAAGAGAAGCATTCTCAGAAACTTCTGAGTGATGATTGCATTCAAGTCACACAGTTGAACCCTCCTTTTGATGGAGCAGTTTTGAAACTGTCTTTTTGTAG
>X02418.1_X11
AATCTGTAAGTGGATACGTGGACCTCTTTGAAGATTTCTTTGGAATCGGGAATATTTCCACAGAAAAACTAAACTGAAGCATTCTCAGAAACTGCTTTGTGGTGTTTGTGTTCGAGCCGCAGAGTTTAACATTGCTTTTCATAGAGCAGTTTTGAAATATTCTTTTGGCAG
>X02418.1_X12
AATCTGCAAGTGGACATTTGGAGCGCTTTCAGGCCTGTGGTGGAAAAGGCTTTATCTTCACAGAAAGACGAGAGAGAAGCATTGTCAGAAACTTCTTTGTGATGATTGCATTCAACTCACAGAGTTGAAGATTCCTTTTGAAACAGCAGTTTCGAAACACTCTTTCTGTGG
```

### Options:

- `-v abovediag=1` — prints index values above diagonal in the top right corner;
- `-v belowdiag=1` — prints index values below diagonal in the bottom left corner;
- `-v similarity=1` — prints percent identity of the two compared sequences in opposed corner.

### Results:

`gawk -v cons=X02418.1-aln-ed1-cons.fas -f ../../considenti.awk X02418.1-aln-ed2.fas`
```
>X02418.1_consX

>X02418.1_X1	-	4	3	22	4	32	0	7	10	4	18	8
>X02418.1_X2	0	-	23	3	8	8	37	17	7	8	7	29
>X02418.1_X3	0	0	-	8	12	0	36	64	8	10	3	29
>X02418.1_X4	0	0	0	-	6	25	3	8	41	13	14	9
>X02418.1_X5	0	0	0	0	-	12	11	13	6	27	7	27
>X02418.1_X6	0	0	0	0	0	-	7	0	14	8	33	8
>X02418.1_X7	0	0	0	0	0	0	-	22	3	4	0	33
>X02418.1_X8	0	0	0	0	0	0	0	-	11	10	9	20
>X02418.1_X9	0	0	0	0	0	0	0	0	-	7	18	10
>X02418.1_X10	0	0	0	0	0	0	0	0	0	-	11	16
>X02418.1_X11	0	0	0	0	0	0	0	0	0	0	-	4
>X02418.1_X12	0	0	0	0	0	0	0	0	0	0	0	-
```

`gawk -v cons=X02418.1-aln-ed1-cons.fas -v belowdiag=1 -f ../../considenti.awk  X02418.1-aln-ed2.fas`
```
>X02418.1_consX

>X02418.1_X1	-	0	0	0	0	0	0	0	0	0	0	0
>X02418.1_X2	4	-	0	0	0	0	0	0	0	0	0	0
>X02418.1_X3	3	23	-	0	0	0	0	0	0	0	0	0
>X02418.1_X4	22	3	8	-	0	0	0	0	0	0	0	0
>X02418.1_X5	4	8	12	6	-	0	0	0	0	0	0	0
>X02418.1_X6	32	8	0	25	12	-	0	0	0	0	0	0
>X02418.1_X7	0	37	36	3	11	7	-	0	0	0	0	0
>X02418.1_X8	7	17	64	8	13	0	22	-	0	0	0	0
>X02418.1_X9	10	7	8	41	6	14	3	11	-	0	0	0
>X02418.1_X10	4	8	10	13	27	8	4	10	7	-	0	0
>X02418.1_X11	18	7	3	14	7	33	0	9	18	11	-	0
>X02418.1_X12	8	29	29	9	27	8	33	20	10	16	4	-
```

`gawk -v cons=X02418.1-aln-ed1-cons.fas -v similarity=1 -f ../../considenti.awk  X02418.1-aln-ed2.fas`
```
>X02418.1_consX

>X02418.1_X1	-	4	3	22	4	32	0	7	10	4	18	8
>X02418.1_X2	75	-	23	3	8	8	37	17	7	8	7	29
>X02418.1_X3	68	75	-	8	12	0	36	64	8	10	3	29
>X02418.1_X4	74	69	66	-	6	25	3	8	41	13	14	9
>X02418.1_X5	75	76	70	67	-	12	11	13	6	27	7	27
>X02418.1_X6	82	77	70	75	77	-	7	0	14	8	33	8
>X02418.1_X7	70	83	78	65	73	75	-	22	3	4	0	33
>X02418.1_X8	71	74	85	69	71	71	73	-	11	10	9	20
>X02418.1_X9	70	72	67	77	70	73	68	68	-	7	18	10
>X02418.1_X10	76	78	70	70	82	77	73	71	72	-	11	16
>X02418.1_X11	74	73	65	67	71	81	69	70	69	75	-	4
>X02418.1_X12	75	82	76	70	79	77	82	75	73	78	71	-
```

`gawk -v cons=X02418.1-aln-ed1-cons.fas -v belowdiag=1 -v similarity=1 -f ../../considenti.awk  X02418.1-aln-ed2.fas`
```
>X02418.1_consX

>X02418.1_X1	-	75	68	74	75	82	70	71	70	76	74	75
>X02418.1_X2	4	-	75	69	76	77	83	74	72	78	73	82
>X02418.1_X3	3	23	-	66	70	70	78	85	67	70	65	76
>X02418.1_X4	22	3	8	-	67	75	65	69	77	70	67	70
>X02418.1_X5	4	8	12	6	-	77	73	71	70	82	71	79
>X02418.1_X6	32	8	0	25	12	-	75	71	73	77	81	77
>X02418.1_X7	0	37	36	3	11	7	-	73	68	73	69	82
>X02418.1_X8	7	17	64	8	13	0	22	-	68	71	70	75
>X02418.1_X9	10	7	8	41	6	14	3	11	-	72	69	73
>X02418.1_X10	4	8	10	13	27	8	4	10	7	-	75	78
>X02418.1_X11	18	7	3	14	7	33	0	9	18	11	-	71
>X02418.1_X12	8	29	29	9	27	8	33	20	10	16	4	-
```
